src/kaltura-player.js

```diff
@@ -14,41 +14,55 @@ import {CastEventType} from './common/cast/cast-event-type';
 import {RemotePlayerManager} from './common/cast/remote-player-manager';
 import {BaseRemotePlayer} from './common/cast/base-remote-player';
 import {RemoteSession} from './common/cast/remote-session';
+import {ControllerProvider, AdsController} from './common/controllers';
+import {BasePlugin} from './common/plugins';
+import {PluginManager} from './common/plugins';
 import {
-  AdsController,
-  BasePlugin,
   Error,
   EventManager,
   EventType as CoreEventType,
+  AdEventType,
   FakeEvent,
   FakeEventTarget,
   loadPlayer,
   TextStyle,
   Track,
-  Utils
+  Utils,
+  registerEngineDecoratorProvider
 } from '@playkit-js/playkit-js';
 
 class KalturaPlayer extends FakeEventTarget {
-  _eventManager: EventManager;
-  _mediaInfo: ?ProviderMediaInfoObject = null;
-  _remotePlayer: ?BaseRemotePlayer = null;
   _localPlayer: Player;
   _provider: Provider;
   _uiWrapper: UIWrapper;
-  _logger: any;
+  _controllerProvider: ControllerProvider;
+  _adsController: ?AdsController;
+  _eventManager: EventManager = new EventManager();
+  _mediaInfo: ?ProviderMediaInfoObject = null;
+  _remotePlayer: ?BaseRemotePlayer = null;
+  _logger: any = getLogger('KalturaPlayer' + Utils.Generator.uniqueId(5));
+  _pluginManager: PluginManager = new PluginManager();
+  _pluginsConfig: KPPluginsConfigObject = {};
+  _pluginsUiComponents: Array<KPUIComponent> = [];
+  _reset: boolean = true;
+  _firstPlay: boolean = true;
+  _playbackStart: boolean = false;
+  _sourceSelected: boolean = false;
 
   constructor(options: KPOptionsObject) {
     super();
-    this._eventManager = new EventManager();
-    const {sources} = options;
+    const {sources, plugins} = options;
     const noSourcesOptions = Utils.Object.mergeDeep({}, options, {sources: null});
+    delete noSourcesOptions.plugins;
     this._localPlayer = loadPlayer(noSourcesOptions);
-    this._logger = getLogger('KalturaPlayer' + Utils.Generator.uniqueId(5));
+    this._controllerProvider = new ControllerProvider(this._pluginManager);
+    this.configure({plugins});
```

Comment:
why do we call configure twice? this.confugure and this._localPlayer.configure calls this._localPlayer.configure

Reply:
the first configs the plugins the second configs the sources